lnwallet/channel.go

```diff
@@ -4354,12 +4345,26 @@ func (lc *LightningChannel) computeView(view *HtlcView,
 	// channel constraints to the final commitment state. If any fee
 	// updates are found in the logs, the commitment fee rate should be
 	// changed, so we'll also set the feePerKw to this new value.
-	filteredHTLCView, uncommitted, err := lc.evaluateHTLCView(
-		view, &ourBalance, &theirBalance, nextHeight, whoseCommitChain,
+	filteredHTLCView, uncommitted, deltas, err := lc.evaluateHTLCView(
+		view, whoseCommitChain, nextHeight,
 	)
 	if err != nil {
 		return 0, 0, 0, nil, err
 	}
+
+	// Add the balance deltas to the balances we got from the commitment
+	// state.
+	if deltas.Local >= 0 {
+		ourBalance += lnwire.MilliSatoshi(deltas.Local)
+	} else {
+		ourBalance -= lnwire.MilliSatoshi(-1 * deltas.Local)
+	}
+	if deltas.Remote >= 0 {
+		theirBalance += lnwire.MilliSatoshi(deltas.Remote)
+	} else {
+		theirBalance -= lnwire.MilliSatoshi(-1 * deltas.Remote)
```

Comment:
Not sure it would be better, but I'd be okay with a struct that expresses increase/decrease for readability:
```
type balanceDelta struct{
    balanceIncrease lnwire.Millisatoshi
    balanceDecrease lnwire.Millisatoshi
}
```

Could also have a `getBalance(currentBal)` method that does the addition + removal so that it's a single line per-party here.

Reply:
In your imagined change here, would you allow both rows to be non-zero, or is this only a struct because go doesn't know what sum types are?

Reply:
Yeah, allow both to be non-zero when we sum up the deltas and then add/subtract each accordingly.
Just seems easier to not deal with negative numbers when we're around `lnwire.Millisatoshi` types.

Reply:
I think I'm going to punt on this change for now unless someone else chimes in saying they want it to work this way. Mostly I just think that having this structure might be more annoying then its worth. We can revisit later if it comes up again.

Reply:
Going to drop this commit completely then?